tasks/migrate-certificates.ts

```diff
@@ -0,0 +1,227 @@
+  MigrateCertificatesTaskOptions,
+  'file'
+>;
+
+export const GET_MIGRATE_CERTIFICATES_TASK = () =>
+  ({
+    name: 'migrate-certificates',
+    description: 'Utility to migrate legacy certificates',
+    run: async (
+      options: MigrateCertificatesTaskOptions,
+      _: CustomHardHatRuntimeEnvironment
+    ): Promise<void> => {
+      const {
+        file,
+        outputFile = 'migrated-certificates.json',
+        dryRun,
+      } = options as ParsedMigrateCertificatesTaskOptions;
+      const jsonData = JSON.parse(readFileSync(file, 'utf8'));
+      // hre.log({ jsonData });
+
+      const [signer] = await hre.getSigners();
+      const signerAddress = await signer.getAddress();
+      const { getRemoval } = await import('@/utils/contracts');
+      const removalContract = await getRemoval({
+        hre,
+        signer,
+      });
+      const certificateContract = await getCertificate({
+        hre,
+        signer,
+      });
+      hre.log(`Removal contract address: ${removalContract.address}`);
+      hre.log(`Certificate contract address: ${certificateContract.address}`);
+      hre.log(`Signer address: ${signerAddress}`);
+      // const fireblocksSigner = removalContract.signer as FireblocksSigner;
+
+      // submit all migrate transactions serially to avoid nonce collision!
+      let certificateIndex = 1;
+      const pendingTransactions = [];
+      if (!dryRun) {
+        hre.log(
+          chalk.bold.white(
+            `✨ Migrating ${jsonData.length} legacy certificates...`
+          )
+        );
+        hre.log(chalk.white(`🤞 Submitting transactions...`));
+      } else {
+        hre.log(
+          chalk.bold.white(
+            `DRY RUN 🌵 Migrating ${jsonData.length} legacy certificates...`
+          )
+        );
+      }
+
+      for (const certificate of jsonData) {
+        const amounts = certificate.amounts.map((amount: string) =>
+          ethers.utils
+            .parseUnits(divide(Number(amount), 1_000_000).toString())
+            .toString()
+        );
+
+        if (!dryRun) {
+          let pendingTx: ethers.ContractTransaction;
+          try {
+            pendingTx = await removalContract.migrate(
+              certificate.ids,
+              amounts,
+              signerAddress, // TODO use Nori admin address, which may also be the signer?
+              ethers.utils
+                .parseUnits(certificate.data.numberOfNrts.toString())
+                .toString()
+            );
+            pendingTransactions.push(pendingTx);
+            hre.log(
+              `(${certificateIndex}/${jsonData.length}) datastore ID: ${
+                certificate.id
+              } txHash: ${chalk.green(pendingTx.hash)}`
+            );
+          } catch (error) {
+            hre.log(
+              `(${certificateIndex}/${jsonData.length}) datastore ID: ${
+                certificate.id
+              } ${chalk.red('error')}`
+            );
+            pendingTransactions.push(error);
```

Comment:
should we throw here instead to guarantee on-chain ordering of certificates?

Reply:
damn, I think I forgot to push my changes from yesterday which does this! I will push whatever I have now

Reply:
(and yes, we should! which the scripts now do)